bin/minifollowups/pycbc_injection_minifollowup

```diff
@@ -14,20 +14,96 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
-""" Followup foreground events """
 
-import os, argparse, numpy, logging, h5py, copy
+"""Create a workflow for following up missed loud injections."""
+
+import os
+import argparse
+import logging
+import copy
+import numpy
+import h5py
 import pycbc.workflow as wf
+import pycbc.workflow.minifollowups as mini
+import pycbc.version
+import pycbc.pnutils
 from pycbc.types import MultiDetOptionAction
 from pycbc.events import select_segments_by_definer, coinc
 from pycbc.results import layout
 from pycbc.detector import Detector
-import pycbc.workflow.minifollowups as mini
 from pycbc.workflow.core import resolve_url_to_file
-import pycbc.version, pycbc.pnutils
 from pycbc.io.hdf import SingleDetTriggers, HFile
+from pycbc.conversions import mchirp_from_mass1_mass2
+
 
-parser = argparse.ArgumentParser(description=__doc__[1:])
+def sort_injections(args, inj_group, missed, sort_by):
+    """Return an array of indices to sort the missed injections according to a
+    metric of choice.
+
+    Parameters
+    ----------
+    args : object
+        CLI arguments parsed by argparse.
+    inj_group : h5py group object
+        HDF5 group object containing the injection definition.
+    missed : array
+        Array of indices of missed injections into `inj_group`.
+    sort_by : str
+        Which metric to sort by.
+
+    Returns
+    -------
+    sorting : array
+        Array of indices that will sort the missed injections as requested.
+    """
+    if 'optimal_snr' in sort_by:
+        optimal_snrs = [
+            inj_group[dsn][:][missed] for dsn in inj_group.keys()
+            if dsn.startswith('optimal_snr_')
+        ]
+        assert optimal_snrs, 'These injections do not have optimal SNRs'
+
+    if sort_by == 'decisive_optimal_snr':
+        # descending order of decisive (2nd largest) optimal SNR
+        dec_snr = numpy.array([
+            sorted(snrs)[-2] for snrs in zip(*optimal_snrs)
+        ])
+        if args.maximum_decisive_snr is not None:
+            # By setting to 0, these injections will not be considered
+            dec_snr[dec_snr > args.maximum_decisive_snr] = 0
+        return dec_snr.argsort()[::-1]
+
+    if sort_by == 'network_optimal_snr':
```

Comment:
Should this use `elif` rather than `if`? I suppose the return means you don't need the `else`

Reply:
Yes, and codeclimate would complain about that.